auth/store.go

```diff
@@ -607,18 +612,26 @@ func (as *authStore) isOpPermitted(userName string, key string, write bool, read
 		}
 
 		for _, perm := range role.KeyPermission {
-			if bytes.Equal(perm.Key, []byte(key)) {
-				if perm.PermType == authpb.READWRITE {
-					return true
+			if rangeEnd == "" {
+				if !bytes.Equal(perm.Key, []byte(key)) {
+					continue
 				}
-
-				if write && !read && perm.PermType == authpb.WRITE {
-					return true
+			} else {
+				if !isSubset([]byte(key), []byte(rangeEnd), perm.Key, perm.RangeEnd) {
```

Comment:
if we have a perm from [a,b) and [b,d), [a,c) should be allowed. but now, it will be denied.